### DISCUSSION TITLE
Resolving a conversation in a review comment deletes draft reply

### DISCUSSION THREAD

**Discussion by qubist** [community/community#23563] [New to GitHub]
<p>Hi there!</p>
<p>Just an unexpected behavior report.</p>
<h2>Steps to reproduce:</h2>
<ol>
<li>I open a PR</li>
<li>Someone reviews and comments on a line of code</li>
<li>I type in a reply but I don’t click “Comment” yet
<ul>
<li><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/0/109d0ed44ee7ae8d646331a7b758261c53fedf9f.png" data-download-href="/uploads/short-url/2mY8ll8MUnHbSi0VtfUpINIHiPR.png?dl=1" title="Screen Shot 2021-04-02 at 12.23.28 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/3607279/181077918-89ed1ec6-1422-473b-a96f-90ebd91f3d82.png" alt="Screen Shot 2021-04-02 at 12.23.28 PM" data-base62-sha1="2mY8ll8MUnHbSi0VtfUpINIHiPR" width="502" height="375" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/0/109d0ed44ee7ae8d646331a7b758261c53fedf9f_2_502x375.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/0/109d0ed44ee7ae8d646331a7b758261c53fedf9f_2_753x562.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/0/109d0ed44ee7ae8d646331a7b758261c53fedf9f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/0/109d0ed44ee7ae8d646331a7b758261c53fedf9f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-04-02 at 12.23.28 PM</span><span class="informations">971×724 50.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div></li>
</ul>
</li>
<li>I click “Resolve conversation” below the reply box</li>
<li>The conversation is resolved and minimized
<ul>
<li>https: [slash slash] i.imgur [dot] com/2DwgNl8.png</li>
</ul>
</li>
<li>I open it back up and my comment is gone
<ul>
<li><a href="https://i.imgur.com/4iaiHAX.png" rel="noopener nofollow ugc">https://i.imgur.com/4iaiHAX.png</a></li>
</ul>
</li>
</ol>
<h2>Expected behavior</h2>
<p>Either I can open up the conversation and my draft reply is still there, or the conversation isn’t minimized when I resolve it but still have text in the “Leave a comment” box. Or maybe I’m not allowed to resolve a conversation while having text in the box? Not sure. But the current behavior caused me to lose work a few times now when writing a comment that also resolves the conversation.</p>
<h2>Details</h2>
<ul>
<li>I was logged in</li>
<li>This happened on an old repo I was using to test (<a href="https://github.com/qubist/aqo/pull/1" class="inline-onebox" rel="noopener nofollow ugc">Create LICENSE by qubist · Pull Request #1 · qubist/aqo · GitHub</a>) as well as other repos.</li>
<li>I don’t think there’s documentation for this.</li>
</ul>
<p><strong>PS:</strong> limitations on number of embedded screenshots and links in a post makes in really hard to post detailed bug reports with pictures</p>

**Comment by ernest-phillips** [comment, marked as answer]
<p>Hello <a class="mention" href="/qubist">@qubist</a> and welcome to the Community!</p>
<p>I agree, that this flow’s intention is not super obvious. May be even a little buggy. I am not sure of the best way to remedy this, but I will bring this up to our design team to get the conversation started.</p>
<p>Thanks for your detailed notes here.</p>

**Comment by fapapa** [comment]
<p>+1 on a fix for this. Personally I’ve been “commenting” and hitting the “resolve conversation” button thinking it was also submitting the comment. Since it minimizes the conversation, I was thinking that my co-workers were getting these comments, but instead they were just seeing me close conversation without doing anything to actually resolve them. Making me look like a jerk.</p>
<p>Disabling the resolve button (or warning me when pressing it) when there’s a draft comment would resolve what is a very bad user experience for me.</p>

**Comment by bhasden** [comment]
<p>I just ran into this issue as well. It’d be great to have some sort of warning even just keeping the draft comment around when clicking “Resolve Conversation”.</p>

**Comment by tfonda-fbk** [comment]
I too would really like to see this fixed. I've just lost an extensive reply I had written under a Discussion answer just because I clicked the "Mark as answer" button before submitting the reply. It really is poor UX.

**Comment by abeehler** [comment]
Had this problem today.  Wrote a long code review comment response and clicked resolve.  Lost comment.  Most other actions will pause the page and ask for confirmation to ensure you don't lose comments.  Perhaps a "resolve with comment" button might be good.  Or just hide the resolve button while commenting.
